src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputAction.java

```diff
@@ -63,31 +62,29 @@ public void onLoad(Run<?, ?> r) {
     private synchronized void loadExecutions() throws InterruptedException, TimeoutException {
         if (executions == null) {
             try {
-            FlowExecution execution = null;
-            for (FlowExecution _execution : FlowExecutionList.get()) {
-                if (_execution.getOwner().getExecutable() == run) {
-                    execution = _execution;
-                    break;
-                }
-            }
-            if (execution != null) {
-                List<StepExecution> candidateExecutions = execution.getCurrentExecutions(true).get(LOAD_EXECUTIONS_TIMEOUT, TimeUnit.SECONDS);
-                executions = new ArrayList<>(); // only set this if we know the answer
-                // JENKINS-37154 sometimes we must block here in order to get accurate results
-                for (StepExecution se : candidateExecutions) {
-                    if (se instanceof InputStepExecution) {
-                        InputStepExecution ise = (InputStepExecution) se;
-                        if (ids.contains(ise.getId())) {
-                            executions.add(ise);
+                if (run instanceof FlowExecutionOwner.Executable) {
+                    var feo = ((FlowExecutionOwner.Executable) run).asFlowExecutionOwner();
+                    if (feo != null) {
+                        var candidateExecutions = feo.get().getCurrentExecutions(true).get(LOAD_EXECUTIONS_TIMEOUT, TimeUnit.SECONDS);
+                        executions = new ArrayList<>(); // only set this if we know the answer
+                        // JENKINS-37154 sometimes we must block here in order to get accurate results
+                        for (StepExecution se : candidateExecutions) {
```

Comment:
FWIW, I have considered a few times adding a `StepExecution.applyAll(FlowExecution e, Function<StepExecution,Void> f)` API method a few times. I think a lot of uses of [`StepExecution.applyAll`](https://github.com/search?type=code&q=+owner%3Ajenkinsci+StepExecution.applyAll) only need to find a step executions in a single build or even just a single execution, and the fact that `FlowExecutionList.StepExecutionIteratorImpl` has to look at all builds can be problematic in various ways (e.g. https://github.com/jenkinsci/workflow-api-plugin/pull/347)

Reply:
https://github.com/jenkinsci/workflow-api-plugin/blob/f906ce67f0230fbed23793199e00820a18ca1581/src/main/java/org/jenkinsci/plugins/workflow/flow/FlowExecutionList.java#L255-L277 essentially, which could be implemented directly in `workflow-step-api` rather than going through `StepExecutionIterator`. Yes that would be helpful. More or less orthogonal to this PR since this is about getting the `FlowExecution` and that would be doing something with it.

Now if https://github.com/jenkinsci/pipeline-plugin/pull/252 had been implemented using `StepExecution.applyAll` that would be another matter!

Reply:
That would be helpful in some cases like https://github.com/jenkinsci/workflow-basic-steps-plugin/blob/2717fe5e51e7c34110610282ccbe70ffd3f0105c/src/main/java/org/jenkinsci/plugins/workflow/steps/WaitForConditionStep.java#L121 where we have a specific `FlowExecution` in mind (or can easily obtain one from a `Run`, `StepContext`, etc.). It would not help in https://github.com/jenkinsci/workflow-durable-task-step-plugin/blob/f6c9e89e5988880ad04890e8ca85c290f79bf929/src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java#L762 or https://github.com/jenkinsci/workflow-basic-steps-plugin/blob/2717fe5e51e7c34110610282ccbe70ffd3f0105c/src/main/java/org/jenkinsci/plugins/workflow/steps/TimeoutStepExecution.java#L309 so I am not sure it is worth the bother.